src/app/components/Navbar.tsx

```diff
@@ -93,10 +122,41 @@ const Navbar = () => {
   );
 };
 
-export default function NavbarComponent() {
+export default Navbar;
+
+const NavHeading = () => {
+  const pathname = usePathname();
+
   return (
-    <Suspense>
-      <Navbar />
-    </Suspense>
+    <div className="flex items-center gap-1">
+      {pathname !== '/' && <BackButton className="text-white" />}
+      <Link href="/">
+        <h5 id="drawer-navigation-label" className="font-bold text-gray-300 uppercase dark:text-gray-400">
+          Dua Collection
+        </h5>
+      </Link>
+    </div>
   );
-}
+};
+
+const MenuLinks = ({ toggle }: INavComponentProps) => {
```

Comment:
Why not create the types specific to these components? e.g. MenuLinksProps interface.